CDCgov/NEDSS-Modernization
# [CNFT2-481] Create Template API

## Description

Adds the POST `/page-builder/api/v1/pages/{page}/template` that allows creating a Template from a Page and ensures that the `name` and `description` are present in the request.  The name of a template must also be unique.

- Adds the `Create Template` request to the Postman collection
- Adds the `NBS Login` request to the Postman collection so that a JSESSIONID cookie can be used by requests that involve interactions with NBS 6.x.
- Adds the `classic-interaction` library to encapsulate interactions with the NBS 6.x.
- Adds the `web` library to provide helper classes for adding, finding, and removing Cookies.
- Adds the `classic-interaction-cucumber` library to support testing interactions with NBS 6.x
- Updates the `auth-cucumber` library to provide an active `SessionCookie` for the `I am logged in` and `I am logged in as "user"` steps 

## Tickets

* [CNFT2-481](https://cdc-nbs.atlassian.net/browse/CNFT2-481)

## Steps to verify

1. Import Postman collection
2. Use the `NBS Login` request to obtain a JSESSIONID from NBS.  (may only need this, if you get a 403 then run the `Login` request to obtain a token.
3. Use the `Pages > Create Page` request to create a new Page
4. Use the `Pages > Create Template` request to create a template from the newly created Page.
5. Verify that the newly created template exists in the `Template Library` by navigating to `System Management > Page Management > Manage Templates`


[CNFT2-481]: https://cdc-nbs.atlassian.net/browse/CNFT2-481?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
